### PR TITLE
add-database

### PR DESCRIPTION
**Pull request by Fiamma-Jikukan**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -15,40 +24,88 @@ app.use("/public", express.static(__dirname + "/public"));
...
+    }
+
 })
 // post requests
 app.post("/signup", async (req, res) => {
+    const signupUser = await User.findOne({ username: req.body.username })
+    try {
+        if (signupUser) {
+            console.log("user not added");
+            res.redirect('/');
```
Comment (MadaraUchiha) [inline server.js:52]: Usually you'd want to let the user know that their username already exists, or, if you want to get *really* fancy, log them in if their password is correct.

File: server.js [hunk 2]
```diff
@@ -15,40 +24,88 @@ app.use("/public", express.static(__dirname + "/public"));
...
+        } else {
+            const newUser = await User.create({
+                username: req.body.username,
+                password: req.body.password,
```
Comment (MadaraUchiha) [inline server.js:56]: Let's do password hashing next

File: server.js [hunk 3]
```diff
@@ -15,40 +24,88 @@ app.use("/public", express.static(__dirname + "/public"));
...
+            });
+            console.log("user added");
+            res.redirect('/');
+        }
 
-    users[req.body.username] = req.body.password;
-    res.redirect('/');
+    } catch (err) {
+        res.status(500).json({ result: 'bad', error: err.message })
```
Comment (MadaraUchiha) [inline server.js]: The same route should not sometimes return HTML and sometimes return JSON. Either you always return JSON (and use client-side JavaScript do do whatever you want with it), or you render an HTML page (in the case of an error, render an error template)

File: server.js [hunk 4]
```diff
@@ -15,40 +24,88 @@ app.use("/public", express.static(__dirname + "/public"));
...
+        if (loginuser) {
+            if (loginuser.password === req.body.password) {
+                const newSession = await Session.create({
+                    user: req.body.username,
+                });
+                console.log(newSession.id);
+                console.log("session added");
+                res.cookie('session_id', newSession.id);
+            } else {
+                res.redirect('/');
```
Comment (MadaraUchiha) [inline server.js:85]: No "your login is bad" error?

File: server.js [hunk 5]
```diff
@@ -15,40 +24,88 @@ app.use("/public", express.static(__dirname + "/public"));
...
+            }
         } else {
-            res.redirect('/');
+            res.render('index', sessions);
```
Comment (MadaraUchiha) [inline server.js]: Why different behavior between when username is wrong and when password is wrong? Both should give an error.

Also, what is the `sessions` object? Aren't you getting your sessions from the DB now?

File: server.js [hunk 6]
```diff
@@ -15,40 +24,88 @@ app.use("/public", express.static(__dirname + "/public"));
...
+app.get("/", async (req, res) => {
     const { cookies } = req
-    res.render('index', sessions[cookies.session_id]);
+    try {
+        const currentSession = await Session.findById(cookies.session_id)
+        if (currentSession === null) {
+            res.render('index');
+        } else {
+            res.render('index', currentSession);
+        }
```
Comment (MadaraUchiha) [inline server.js]: What's the difference between passing `currentSession` and not passing it? Aren't you able to do just
```suggestion
        res.render('index', currentSession);
```
And if it's null it's null?

**Review by MadaraUchiha — COMMENTED** [review]
Needs a bit of cleanup here and there, but overall looks good.

File: server.js (hunk 5 above)
Comment (Fiamma-Jikukan) [reply server.js]: both are not working for some reason. The server crushes and gives: ErrorCaptureStackTrace(err);

**Pull request merged**